src/pip/_internal/req/req_install.py

```diff
@@ -627,6 +627,8 @@ def prepare_pep517_metadata(self):
 
         self.metadata_directory = os.path.join(metadata_dir, distinfo_dir)
 
+    # NOTE: This is only used in get_dist(), so look into eagerly assigning to
```

Comment:
Can we make an issue for this instead of noting it here? Or we can find some other way to track the next easiest bit of refactoring or ideas for other reorganization that are blocked by another action.

Reply:
Done! Moved it over to a project card. :)